### PR TITLE
Shadcn migration - search modal

### PR DESCRIPTION
**Pull request by pettinarip**
## Description

Search modal implementation using Shadcn.

### CODE REVIEW

File: src/components/Search/utils.ts
Comment (pettinarip) [inline src/components/Search/utils.ts:1]: All this now lives in `src/styles/docsearch.css`.

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -6,8 +6,10 @@ import { AppPropsWithLayout } from "@/lib/types"
 
 import ThemeProvider from "@/components/ThemeProvider"
 
+import "@docsearch/css"
 import "@/styles/global.css"
 import "@/styles/fonts.css"
+import "@/styles/docsearch.css"
```
Comment (pettinarip) [inline src/pages/_app.tsx:12]: Changed the `eslint` import sort config to allow this. The order of these imports is important to override the external styles with our custom styles.

File: package.json [hunk 2]
```diff
@@ -69,6 +70,7 @@
     "react-i18next": "^13.3.1",
     "react-icons": "^4.10.1",
     "react-lite-youtube-embed": "^2.4.0",
+    "react-merge-refs": "^2.1.1",
```
Comment (pettinarip) [inline package.json:73]: Added a replacement of `useMergeRefs` from Chakra.

File: .eslintrc.json [hunk 3]
```diff
@@ -26,55 +22,29 @@
...
-            "^\\./(?=.*/)(?!/?$)",
-            "^\\.(?!/?$)",
-            "^\\./?$"
-          ],
+          ["^\\./(?=.*/)(?!/?$)", "^\\.(?!/?$)", "^\\./?$"],
           // Style imports.
-          [
-            "^.+\\.s?css$"
-          ],
+          ["^.+s?css$"],
```
[the remark this answers is not in the document]
Comment (wackerow) [reply .eslintrc.json:45]: Works for me. If we discover false positives we can address from there

**Review by wackerow — APPROVED** [review]
Working great, thanks @pettinarip!

**Comment by Wachira3712** [comment]
ครับ

**Pull request merged**